controller/controller_server/csi_controller_server.py

```diff
@@ -388,10 +388,43 @@ def ControllerUnpublishVolume(self, request, context):
             return csi_pb2.ControllerUnpublishVolumeResponse()
 
     def ValidateVolumeCapabilities(self, request, context):
+        set_current_thread_name(request.volume_id)
         logger.info("ValidateVolumeCapabilities")
-        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
-        logger.info("finished ValidateVolumeCapabilities")
-        return csi_pb2.ValidateVolumeCapabilitiesResponse()
+        try:
+
+            utils.validate_validate_volume_capabilities_request(request)
+
+            secrets = request.secrets
+            user, password, array_addresses = utils.get_array_connection_info_from_secret(secrets)
+
+            # TODO : pass multiple array addresses
+            array_type, volume_id = utils.get_volume_id_info(request.volume_id)
+
+            with get_agent(user, password, array_addresses, array_type).get_mediator() as array_mediator:
+                try:
+                    volume = array_mediator.get_volume_by_id(volume_id=volume_id)
+                except controller_errors.VolumeNotFoundError as ex:
```

Comment:
isn't it already excepted in [line 418](https://github.com/IBM/ibm-block-csi-driver/blob/cc133625139db6729e4b2dbd582eb8de2c70eb38/controller/controller_server/csi_controller_server.py#L418)?

Reply:
yep, removed